### ISSUE TITLE
`[WARN] serf: Query queue depth: 3216` messages in the logs

### ISSUE THREAD

**Issue by davidgengenbach** [distribworks/dkron#649]
**Describe the bug**
We get a lot of `serf` warnings, e.g. `[WARN] serf: Query queue depth: 3216` where the queue depth increases steadily (~ queue depth increases at a rate of 2 per minute).

<details>
  <summary>Logs</summary>

  ```´´´
  time="2019-11-30T10:20:14Z" level=info msg="2019/11/30 10:20:14 [WARN] serf: Query queue depth: 3216"
  time="2019-11-30T10:20:44Z" level=info msg="2019/11/30 10:20:44 [WARN] serf: Query queue depth: 3216"
  time="2019-11-30T10:21:14Z" level=info msg="2019/11/30 10:21:14 [WARN] serf: Query queue depth: 3218"
  time="2019-11-30T10:21:44Z" level=info msg="2019/11/30 10:21:44 [WARN] serf: Query queue depth: 3218"
  time="2019-11-30T10:22:14Z" level=info msg="2019/11/30 10:22:14 [WARN] serf: Query queue depth: 3220"
  time="2019-11-30T10:22:44Z" level=info msg="2019/11/30 10:22:44 [WARN] serf: Query queue depth: 3220"
  time="2019-11-30T10:23:14Z" level=info msg="2019/11/30 10:23:14 [WARN] serf: Query queue depth: 3222"
  time="2019-11-30T10:23:44Z" level=info msg="2019/11/30 10:23:44 [WARN] serf: Query queue depth: 3222"
  time="2019-11-30T10:24:14Z" level=info msg="2019/11/30 10:24:14 [WARN] serf: Query queue depth: 3224"
  time="2019-11-30T10:24:44Z" level=info msg="2019/11/30 10:24:44 [WARN] serf: Query queue depth: 3224"
  time="2019-11-30T10:25:14Z" level=info msg="2019/11/30 10:25:14 [WARN] serf: Query queue depth: 3227"
  time="2019-11-30T10:25:44Z" level=info msg="2019/11/30 10:25:44 [WARN] serf: Query queue depth: 3227"
  time="2019-11-30T10:26:14Z" level=info msg="2019/11/30 10:26:14 [WARN] serf: Query queue depth: 3229"
  time="2019-11-30T10:26:44Z" level=info msg="2019/11/30 10:26:44 [WARN] serf: Query queue depth: 3229"
  time="2019-11-30T10:27:14Z" level=info msg="2019/11/30 10:27:14 [WARN] serf: Query queue depth: 3231"
  time="2019-11-30T10:27:44Z" level=info msg="2019/11/30 10:27:44 [WARN] serf: Query queue depth: 3231"
  time="2019-11-30T10:28:14Z" level=info msg="2019/11/30 10:28:14 [WARN] serf: Query queue depth: 3233"
  time="2019-11-30T10:28:44Z" level=info msg="2019/11/30 10:28:44 [WARN] serf: Query queue depth: 3233"
  time="2019-11-30T10:29:14Z" level=info msg="2019/11/30 10:29:14 [WARN] serf: Query queue depth: 3235"
  time="2019-11-30T10:29:44Z" level=info msg="2019/11/30 10:29:44 [WARN] serf: Query queue depth: 3235"
  time="2019-11-30T10:30:14Z" level=info msg="2019/11/30 10:30:14 [WARN] serf: Query queue depth: 3239"
  time="2019-11-30T10:30:44Z" level=info msg="2019/11/30 10:30:44 [WARN] serf: Query queue depth: 3239"
  time="2019-11-30T10:31:14Z" level=info msg="2019/11/30 10:31:14 [WARN] serf: Query queue depth: 3241"
  time="2019-11-30T10:31:44Z" level=info msg="2019/11/30 10:31:44 [WARN] serf: Query queue depth: 3241"
  time="2019-11-30T10:32:14Z" level=info msg="2019/11/30 10:32:14 [WARN] serf: Query queue depth: 3243"
  time="2019-11-30T10:32:44Z" level=info msg="2019/11/30 10:32:44 [WARN] serf: Query queue depth: 3243"
  time="2019-11-30T10:33:14Z" level=info msg="2019/11/30 10:33:14 [WARN] serf: Query queue depth: 3245"
  time="2019-11-30T10:33:44Z" level=info msg="2019/11/30 10:33:44 [WARN] serf: Query queue depth: 3245"
  time="2019-11-30T10:34:14Z" level=info msg="2019/11/30 10:34:14 [WARN] serf: Query queue depth: 3247"
  time="2019-11-30T10:34:44Z" level=info msg="2019/11/30 10:34:44 [WARN] serf: Query queue depth: 3247"
  time="2019-11-30T10:35:14Z" level=info msg="2019/11/30 10:35:14 [WARN] serf: Query queue depth: 3250"
  time="2019-11-30T10:35:44Z" level=info msg="2019/11/30 10:35:44 [WARN] serf: Query queue depth: 3250"
  time="2019-11-30T10:36:14Z" level=info msg="2019/11/30 10:36:14 [WARN] serf: Query queue depth: 3252"
  time="2019-11-30T10:36:44Z" level=info msg="2019/11/30 10:36:44 [WARN] serf: Query queue depth: 3252"
  time="2019-11-30T10:37:14Z" level=info msg="2019/11/30 10:37:14 [WARN] serf: Query queue depth: 3254"
  time="2019-11-30T10:37:44Z" level=info msg="2019/11/30 10:37:44 [WARN] serf: Query queue depth: 3254"
  time="2019-11-30T10:38:14Z" level=info msg="2019/11/30 10:38:14 [WARN] serf: Query queue depth: 3256"
  time="2019-11-30T10:38:44Z" level=info msg="2019/11/30 10:38:44 [WARN] serf: Query queue depth: 3256"
  time="2019-11-30T10:39:14Z" level=info msg="2019/11/30 10:39:14 [WARN] serf: Query queue depth: 3258"
  time="2019-11-30T10:39:44Z" level=info msg="2019/11/30 10:39:44 [WARN] serf: Query queue depth: 3258"
  time="2019-11-30T10:40:14Z" level=info msg="2019/11/30 10:40:14 [WARN] serf: Query queue depth: 3261"
  time="2019-11-30T10:40:44Z" level=info msg="2019/11/30 10:40:44 [WARN] serf: Query queue depth: 3261"
  time="2019-11-30T10:41:14Z" level=info msg="2019/11/30 10:41:14 [WARN] serf: Query queue depth: 3263"
  time="2019-11-30T10:41:44Z" level=info msg="2019/11/30 10:41:44 [WARN] serf: Query queue depth: 3263"
  time="2019-11-30T10:42:14Z" level=info msg="2019/11/30 10:42:14 [WARN] serf: Query queue depth: 3265"
  time="2019-11-30T10:42:44Z" level=info msg="2019/11/30 10:42:44 [WARN] serf: Query queue depth: 3265"
  time="2019-11-30T10:43:14Z" level=info msg="2019/11/30 10:43:14 [WARN] serf: Query queue depth: 3267"
  time="2019-11-30T10:43:44Z" level=info msg="2019/11/30 10:43:44 [WARN] serf: Query queue depth: 3267"
  time="2019-11-30T10:44:14Z" level=info msg="2019/11/30 10:44:14 [WARN] serf: Query queue depth: 3269"
  time="2019-11-30T10:44:44Z" level=info msg="2019/11/30 10:44:44 [WARN] serf: Query queue depth: 3269"
  time="2019-11-30T10:45:14Z" level=info msg="2019/11/30 10:45:14 [WARN] serf: Query queue depth: 3272"
  time="2019-11-30T10:45:44Z" level=info msg="2019/11/30 10:45:44 [WARN] serf: Query queue depth: 3272"
  time="2019-11-30T10:46:14Z" level=info msg="2019/11/30 10:46:14 [WARN] serf: Query queue depth: 3274"
  time="2019-11-30T10:46:44Z" level=info msg="2019/11/30 10:46:44 [WARN] serf: Query queue depth: 3274"
  time="2019-11-30T10:47:14Z" level=info msg="2019/11/30 10:47:14 [WARN] serf: Query queue depth: 3276"
  time="2019-11-30T10:47:44Z" level=info msg="2019/11/30 10:47:44 [WARN] serf: Query queue depth: 3276"
  time="2019-11-30T10:48:14Z" level=info msg="2019/11/30 10:48:14 [WARN] serf: Query queue depth: 3278"
  time="2019-11-30T10:48:44Z" level=info msg="2019/11/30 10:48:44 [WARN] serf: Query queue depth: 3278"
  time="2019-11-30T10:49:14Z" level=info msg="2019/11/30 10:49:14 [WARN] serf: Query queue depth: 3280"
  time="2019-11-30T10:49:44Z" level=info msg="2019/11/30 10:49:44 [WARN] serf: Query queue depth: 3280"
  time="2019-11-30T10:50:14Z" level=info msg="2019/11/30 10:50:14 [WARN] serf: Query queue depth: 3283"
  time="2019-11-30T10:50:44Z" level=info msg="2019/11/30 10:50:44 [WARN] serf: Query queue depth: 3283"
  time="2019-11-30T10:51:14Z" level=info msg="2019/11/30 10:51:14 [WARN] serf: Query queue depth: 3285"
  time="2019-11-30T10:51:44Z" level=info msg="2019/11/30 10:51:44 [WARN] serf: Query queue depth: 3285"
  time="2019-11-30T10:52:14Z" level=info msg="2019/11/30 10:52:14 [WARN] serf: Query queue depth: 3287"
  time="2019-11-30T10:52:44Z" level=info msg="2019/11/30 10:52:44 [WARN] serf: Query queue depth: 3287"
  time="2019-11-30T10:53:14Z" level=info msg="2019/11/30 10:53:14 [WARN] serf: Query queue depth: 3289"
  time="2019-11-30T10:53:44Z" level=info msg="2019/11/30 10:53:44 [WARN] serf: Query queue depth: 3289"
  time="2019-11-30T10:54:14Z" level=info msg="2019/11/30 10:54:14 [WARN] serf: Query queue depth: 3291"
  time="2019-11-30T10:54:44Z" level=info msg="2019/11/30 10:54:44 [WARN] serf: Query queue depth: 3291"
  time="2019-11-30T10:55:14Z" level=info msg="2019/11/30 10:55:14 [WARN] serf: Query queue depth: 3294"
  time="2019-11-30T10:55:44Z" level=info msg="2019/11/30 10:55:44 [WARN] serf: Query queue depth: 3294"
  time="2019-11-30T10:56:14Z" level=info msg="2019/11/30 10:56:14 [WARN] serf: Query queue depth: 3296"
  time="2019-11-30T10:56:44Z" level=info msg="2019/11/30 10:56:44 [WARN] serf: Query queue depth: 3296"
  time="2019-11-30T10:57:14Z" level=info msg="2019/11/30 10:57:14 [WARN] serf: Query queue depth: 3298"
  time="2019-11-30T10:57:44Z" level=info msg="2019/11/30 10:57:44 [WARN] serf: Query queue depth: 3298"
  time="2019-11-30T10:58:14Z" level=info msg="2019/11/30 10:58:14 [WARN] serf: Query queue depth: 3300"
  time="2019-11-30T10:58:44Z" level=info msg="2019/11/30 10:58:44 [WARN] serf: Query queue depth: 3300"
  time="2019-11-30T10:59:14Z" level=info msg="2019/11/30 10:59:14 [WARN] serf: Query queue depth: 3302"
  time="2019-11-30T10:59:44Z" level=info msg="2019/11/30 10:59:44 [WARN] serf: Query queue depth: 3302"
  time="2019-11-30T11:00:14Z" level=info msg="2019/11/30 11:00:14 [WARN] serf: Query queue depth: 3306"
  time="2019-11-30T11:00:44Z" level=info msg="2019/11/30 11:00:44 [WARN] serf: Query queue depth: 3306"
  time="2019-11-30T11:01:14Z" level=info msg="2019/11/30 11:01:14 [WARN] serf: Query queue depth: 3308"
  time="2019-11-30T11:01:44Z" level=info msg="2019/11/30 11:01:44 [WARN] serf: Query queue depth: 3308"
  time="2019-11-30T11:02:14Z" level=info msg="2019/11/30 11:02:14 [WARN] serf: Query queue depth: 3310"
  time="2019-11-30T11:02:44Z" level=info msg="2019/11/30 11:02:44 [WARN] serf: Query queue depth: 3310"
  time="2019-11-30T11:03:14Z" level=info msg="2019/11/30 11:03:14 [WARN] serf: Query queue depth: 3312"
  time="2019-11-30T11:03:44Z" level=info msg="2019/11/30 11:03:44 [WARN] serf: Query queue depth: 3312"
  time="2019-11-30T11:04:14Z" level=info msg="2019/11/30 11:04:14 [WARN] serf: Query queue depth: 3314"
  time="2019-11-30T11:04:44Z" level=info msg="2019/11/30 11:04:44 [WARN] serf: Query queue depth: 3314"
  time="2019-11-30T11:05:14Z" level=info msg="2019/11/30 11:05:14 [WARN] serf: Query queue depth: 3317"
  time="2019-11-30T11:05:44Z" level=info msg="2019/11/30 11:05:44 [WARN] serf: Query queue depth: 3317"
  time="2019-11-30T11:06:14Z" level=info msg="2019/11/30 11:06:14 [WARN] serf: Query queue depth: 3319"
  time="2019-11-30T11:06:44Z" level=info msg="2019/11/30 11:06:44 [WARN] serf: Query queue depth: 3319"
  time="2019-11-30T11:07:14Z" level=info msg="2019/11/30 11:07:14 [WARN] serf: Query queue depth: 3321"
  time="2019-11-30T11:07:44Z" level=info msg="2019/11/30 11:07:44 [WARN] serf: Query queue depth: 3321"
  time="2019-11-30T11:08:14Z" level=info msg="2019/11/30 11:08:14 [WARN] serf: Query queue depth: 3323"
  time="2019-11-30T11:08:44Z" level=info msg="2019/11/30 11:08:44 [WARN] serf: Query queue depth: 3323"
  time="2019-11-30T11:09:14Z" level=info msg="2019/11/30 11:09:14 [WARN] serf: Query queue depth: 3325"
  time="2019-11-30T11:09:44Z" level=info msg="2019/11/30 11:09:44 [WARN] serf: Query queue depth: 3325"
  time="2019-11-30T11:10:14Z" level=info msg="2019/11/30 11:10:14 [WARN] serf: Query queue depth: 3328"
  time="2019-11-30T11:10:44Z" level=info msg="2019/11/30 11:10:44 [WARN] serf: Query queue depth: 3328"
  time="2019-11-30T11:11:14Z" level=info msg="2019/11/30 11:11:14 [WARN] serf: Query queue depth: 3330"
  time="2019-11-30T11:11:44Z" level=info msg="2019/11/30 11:11:44 [WARN] serf: Query queue depth: 3330"
  time="2019-11-30T11:12:14Z" level=info msg="2019/11/30 11:12:14 [WARN] serf: Query queue depth: 3332"
  time="2019-11-30T11:12:44Z" level=info msg="2019/11/30 11:12:44 [WARN] serf: Query queue depth: 3332"
  time="2019-11-30T11:13:14Z" level=info msg="2019/11/30 11:13:14 [WARN] serf: Query queue depth: 3334"
  time="2019-11-30T11:13:44Z" level=info msg="2019/11/30 11:13:44 [WARN] serf: Query queue depth: 3334"
  time="2019-11-30T11:14:14Z" level=info msg="2019/11/30 11:14:14 [WARN] serf: Query queue depth: 3336"
  time="2019-11-30T11:14:44Z" level=info msg="2019/11/30 11:14:44 [WARN] serf: Query queue depth: 3336"
  time="2019-11-30T11:15:14Z" level=info msg="2019/11/30 11:15:14 [WARN] serf: Query queue depth: 3339"
  time="2019-11-30T11:15:44Z" level=info msg="2019/11/30 11:15:44 [WARN] serf: Query queue depth: 3339"
  time="2019-11-30T11:16:14Z" level=info msg="2019/11/30 11:16:14 [WARN] serf: Query queue depth: 3341"
  time="2019-11-30T11:16:44Z" level=info msg="2019/11/30 11:16:44 [WARN] serf: Query queue depth: 3341"
  time="2019-11-30T11:17:14Z" level=info msg="2019/11/30 11:17:14 [WARN] serf: Query queue depth: 3343"
  time="2019-11-30T11:17:44Z" level=info msg="2019/11/30 11:17:44 [WARN] serf: Query queue depth: 3343"
  time="2019-11-30T11:18:14Z" level=info msg="2019/11/30 11:18:14 [WARN] serf: Query queue depth: 3345"
  time="2019-11-30T11:18:44Z" level=info msg="2019/11/30 11:18:44 [WARN] serf: Query queue depth: 3345"
  time="2019-11-30T11:19:14Z" level=info msg="2019/11/30 11:19:14 [WARN] serf: Query queue depth: 3347"
  time="2019-11-30T11:19:44Z" level=info msg="2019/11/30 11:19:44 [WARN] serf: Query queue depth: 3347"
  time="2019-11-30T11:20:14Z" level=info msg="2019/11/30 11:20:14 [WARN] serf: Query queue depth: 3350"
  time="2019-11-30T11:20:44Z" level=info msg="2019/11/30 11:20:44 [WARN] serf: Query queue depth: 3350"
  time="2019-11-30T11:21:14Z" level=info msg="2019/11/30 11:21:14 [WARN] serf: Query queue depth: 3352"
  time="2019-11-30T11:21:44Z" level=info msg="2019/11/30 11:21:44 [WARN] serf: Query queue depth: 3352"
  time="2019-11-30T11:22:14Z" level=info msg="2019/11/30 11:22:14 [WARN] serf: Query queue depth: 3354"
  time="2019-11-30T11:22:44Z" level=info msg="2019/11/30 11:22:44 [WARN] serf: Query queue depth: 3354"
  time="2019-11-30T11:23:14Z" level=info msg="2019/11/30 11:23:14 [WARN] serf: Query queue depth: 3356"
  time="2019-11-30T11:23:44Z" level=info msg="2019/11/30 11:23:44 [WARN] serf: Query queue depth: 3356"
  time="2019-11-30T11:24:14Z" level=info msg="2019/11/30 11:24:14 [WARN] serf: Query queue depth: 3358"
  time="2019-11-30T11:24:44Z" level=info msg="2019/11/30 11:24:44 [WARN] serf: Query queue depth: 3358"
  time="2019-11-30T11:25:14Z" level=info msg="2019/11/30 11:25:14 [WARN] serf: Query queue depth: 3361"
  time="2019-11-30T11:25:44Z" level=info msg="2019/11/30 11:25:44 [WARN] serf: Query queue depth: 3361"
  time="2019-11-30T11:26:14Z" level=info msg="2019/11/30 11:26:14 [WARN] serf: Query queue depth: 3363"
  time="2019-11-30T11:26:44Z" level=info msg="2019/11/30 11:26:44 [WARN] serf: Query queue depth: 3363"
  time="2019-11-30T11:27:14Z" level=info msg="2019/11/30 11:27:14 [WARN] serf: Query queue depth: 3365"
  time="2019-11-30T11:27:44Z" level=info msg="2019/11/30 11:27:44 [WARN] serf: Query queue depth: 3365"
  time="2019-11-30T11:28:14Z" level=info msg="2019/11/30 11:28:14 [WARN] serf: Query queue depth: 3367"
  time="2019-11-30T11:28:44Z" level=info msg="2019/11/30 11:28:44 [WARN] serf: Query queue depth: 3367"
  time="2019-11-30T11:29:14Z" level=info msg="2019/11/30 11:29:14 [WARN] serf: Query queue depth: 3369"
  time="2019-11-30T11:29:44Z" level=info msg="2019/11/30 11:29:44 [WARN] serf: Query queue depth: 3369"
  time="2019-11-30T11:30:14Z" level=info msg="2019/11/30 11:30:14 [WARN] serf: Query queue depth: 3373"
  time="2019-11-30T11:30:44Z" level=info msg="2019/11/30 11:30:44 [WARN] serf: Query queue depth: 3373"
  time="2019-11-30T11:31:14Z" level=info msg="2019/11/30 11:31:14 [WARN] serf: Query queue depth: 3375"
  time="2019-11-30T11:31:44Z" level=info msg="2019/11/30 11:31:44 [WARN] serf: Query queue depth: 3375"
  time="2019-11-30T11:32:14Z" level=info msg="2019/11/30 11:32:14 [WARN] serf: Query queue depth: 3377"
  time="2019-11-30T11:32:44Z" level=info msg="2019/11/30 11:32:44 [WARN] serf: Query queue depth: 3377"
  time="2019-11-30T11:33:14Z" level=info msg="2019/11/30 11:33:14 [WARN] serf: Query queue depth: 3379"
  time="2019-11-30T11:33:44Z" level=info msg="2019/11/30 11:33:44 [WARN] serf: Query queue depth: 3379"
  time="2019-11-30T11:34:14Z" level=info msg="2019/11/30 11:34:14 [WARN] serf: Query queue depth: 3381"
  time="2019-11-30T11:34:44Z" level=info msg="2019/11/30 11:34:44 [WARN] serf: Query queue depth: 3381"
  time="2019-11-30T11:35:14Z" level=info msg="2019/11/30 11:35:14 [WARN] serf: Query queue depth: 3384"
  time="2019-11-30T11:35:44Z" level=info msg="2019/11/30 11:35:44 [WARN] serf: Query queue depth: 3384"
  time="2019-11-30T11:36:14Z" level=info msg="2019/11/30 11:36:14 [WARN] serf: Query queue depth: 3386"
  time="2019-11-30T11:36:44Z" level=info msg="2019/11/30 11:36:44 [WARN] serf: Query queue depth: 3386"
  time="2019-11-30T11:37:14Z" level=info msg="2019/11/30 11:37:14 [WARN] serf: Query queue depth: 3388"
  time="2019-11-30T11:37:44Z" level=info msg="2019/11/30 11:37:44 [WARN] serf: Query queue depth: 3388"
  time="2019-11-30T11:38:14Z" level=info msg="2019/11/30 11:38:14 [WARN] serf: Query queue depth: 3390"
  time="2019-11-30T11:38:44Z" level=info msg="2019/11/30 11:38:44 [WARN] serf: Query queue depth: 3390"
  time="2019-11-30T11:39:14Z" level=info msg="2019/11/30 11:39:14 [WARN] serf: Query queue depth: 3392"
  time="2019-11-30T11:39:44Z" level=info msg="2019/11/30 11:39:44 [WARN] serf: Query queue depth: 3392"
  time="2019-11-30T11:40:14Z" level=info msg="2019/11/30 11:40:14 [WARN] serf: Query queue depth: 3395"
  time="2019-11-30T11:40:44Z" level=info msg="2019/11/30 11:40:44 [WARN] serf: Query queue depth: 3395"
  time="2019-11-30T11:41:14Z" level=info msg="2019/11/30 11:41:14 [WARN] serf: Query queue depth: 3397"
  time="2019-11-30T11:41:44Z" level=info msg="2019/11/30 11:41:44 [WARN] serf: Query queue depth: 3397"
  time="2019-11-30T11:42:14Z" level=info msg="2019/11/30 11:42:14 [WARN] serf: Query queue depth: 3399"
  time="2019-11-30T11:42:44Z" level=info msg="2019/11/30 11:42:44 [WARN] serf: Query queue depth: 3399"
  time="2019-11-30T11:43:14Z" level=info msg="2019/11/30 11:43:14 [WARN] serf: Query queue depth: 3401"
  time="2019-11-30T11:43:44Z" level=info msg="2019/11/30 11:43:44 [WARN] serf: Query queue depth: 3401"
  time="2019-11-30T11:44:14Z" level=info msg="2019/11/30 11:44:14 [WARN] serf: Query queue depth: 3403"
  time="2019-11-30T11:44:44Z" level=info msg="2019/11/30 11:44:44 [WARN] serf: Query queue depth: 3403"
  time="2019-11-30T11:45:14Z" level=info msg="2019/11/30 11:45:14 [WARN] serf: Query queue depth: 3406"
  time="2019-11-30T11:45:44Z" level=info msg="2019/11/30 11:45:44 [WARN] serf: Query queue depth: 3406"
  time="2019-11-30T11:46:14Z" level=info msg="2019/11/30 11:46:14 [WARN] serf: Query queue depth: 3408"
  time="2019-11-30T11:46:44Z" level=info msg="2019/11/30 11:46:44 [WARN] serf: Query queue depth: 3408"
  time="2019-11-30T11:47:14Z" level=info msg="2019/11/30 11:47:14 [WARN] serf: Query queue depth: 3410"
  time="2019-11-30T11:47:44Z" level=info msg="2019/11/30 11:47:44 [WARN] serf: Query queue depth: 3410"
  time="2019-11-30T11:48:14Z" level=info msg="2019/11/30 11:48:14 [WARN] serf: Query queue depth: 3412"
  time="2019-11-30T11:48:44Z" level=info msg="2019/11/30 11:48:44 [WARN] serf: Query queue depth: 3412"
  time="2019-11-30T11:49:14Z" level=info msg="2019/11/30 11:49:14 [WARN] serf: Query queue depth: 3414"
  time="2019-11-30T11:49:44Z" level=info msg="2019/11/30 11:49:44 [WARN] serf: Query queue depth: 3414"
  time="2019-11-30T11:50:14Z" level=info msg="2019/11/30 11:50:14 [WARN] serf: Query queue depth: 3417"
  time="2019-11-30T11:50:44Z" level=info msg="2019/11/30 11:50:44 [WARN] serf: Query queue depth: 3417"
  ```

</details>

The complete (redacted) logs are here: 
[cleaned.txt](https://github.com/distribworks/dkron/files/3907099/cleaned.txt)

**To Reproduce**
1. Create jobs
2. Let them run for a day or two

**Expected behavior**
These warnings should not happen?

**Specifications:**
 - OS: Docker on Kubernetes
    - Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-13T11:23:11Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
    - Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.10", GitCommit:"37d169313237cb4ceb2cc4bef300f2ae3053c1a2", GitTreeState:"clean", BuildDate:"2019-08-19T10:44:49Z", GoVersion:"go1.11.13", Compiler:"gc", Platform:"linux/amd64"}
 - Dkron 2.0.0, Agent Protocol: 5 (Understands back to: 2)
   - using the `dkron/dkron:v2` Docker image

**Additional context**
I found an old [StackOverflow post](https://stackoverflow.com/questions/54686388/serf-query-queue-depth-warning-endless-counting-up-no-documentation-on-source) with the same question but not much else.

This is happening a single-server configuration with `--bootstrap-expect 1`.

Is this something to be worried about or is it normal?

**Thousand thanks** :+1:

**Comment by victorcoder** [comment]
@davidgengenbach I've never observed this on moderate workloads, how many jobs and with what frequency are you running?

**Comment by davidgengenbach** [comment]
> @davidgengenbach I've never observed this on moderate workloads, how many jobs and with what frequency are you running?

Jobs:
- `0 * * * * ?` (every minute)
- `0 */5 * * * ?` (every five minutes)
- `0 */5 * * * ?`

Happens in a single-server configuration. The number somehow stabilized at a queue depth of ~ 4100.

As we are evaluating using Dkron for an important part of our infrastructure we are especially keen on having it running perfectly.

We might test multi-server configurations but hoped that a single server suffices as we have a microservice interfacing Dkron and do not need that much failure tolerance of Dkron.
Multiserver configurations might complicate our architecture a little.

**Comment by victorcoder** [comment]
That's not too much, check the test instances I have at test.dkron.io:8080 I try to maintain it with a good amount of workload, and doesn't get the queue issue you are experiencing.

I think you are experiencing something network related.

Could you paste here your `dkron.yml` config file?

**Comment by davidgengenbach** [comment]
> That's not too much, check the test instances I have at test.dkron.io:8080 I try to maintain it with a good amount of workload, and doesn't get the queue issue you are experiencing.

The Dkron test setup is not a single-server configuration - maybe it's related to that? Also, I think the messages only show when having `log-level: debug`.

> I think you are experiencing something network related.

We have a single-server configuration and - I guess - only loopback networking in serf?

> Could you paste here your dkron.yml config file?
```yaml
server: true
log-level: debug
bootstrap-expect: 1
webhook-headers: 'Content-Type: application/json'
webhook-url: 'http://REDACTED'
webhook-payload: >-
  {
    "name": "{{.JobName}}",
    "success": {{.Success}},
    "finishedAt": "{{.FinishedAt}}",
    "startedAt": "{{.StartTime}}",
    "reportingNode": "{{.ReportingNode}}",
    "nodeName": "{{.NodeName}}"
  }
```

I will observe the issue further!

**Comment by victorcoder** [comment]
> Your Dkron test setup is not a single-server configuration - maybe it's related to that? Also, I think the messages only show when having log-level: debug.

Single server config is actually much simpler than the test instance and should give you better performance because of less network roundtrips.

Test instance use debug log level https://github.com/distribworks/dkron/blob/master/scripts/ansible/site.yml#L33

> We have a single-server configuration and - I guess - only loopback networking in serf?

What's the server serf IP address? you can check it in the UI

The config is super simple it should work.

**Comment by victorcoder** [comment]
Fixed in #749
